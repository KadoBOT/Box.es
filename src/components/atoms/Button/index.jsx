import styled from 'styled-components';

const Button = styled.button`
  width: ${props => props.follow ? '8em' : '100%'};
  height: ${props => props.follow ? '30px' : '45px'};
  background: ${props => props.follow ? 'transparent' : '#09b509'};;
  outline: none;
  border: ${props => props.follow ? '1px solid black' : 'none'};;
  border-radius: ${props => props.follow ? '3px' : '6px'};
  float: right;
  color: ${props => props.follow ? 'black' : 'white'};
  margin-top: 5px;
  font-size: ${props => props.follow ? 'small' : '17px'};
  display: flex;
  align-items: baseline;
  justify-content: center;
  margin-right: ${props => props.follow ? '17px': '0'}
`

export default Button
