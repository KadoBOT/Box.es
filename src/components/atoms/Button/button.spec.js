import React from 'react'
import renderer from 'react-test-renderer';
import Button from './index'

it('should render the Button', () => {
    const wrapper = renderer.create(
        <Button />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
});

it('should render the Follow Button', () => {
    const wrapper = renderer.create(
        <Button follow/>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
});
