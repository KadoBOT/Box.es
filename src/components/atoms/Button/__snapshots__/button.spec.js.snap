exports[`test should render the Button 1`] = `
<button
  className="ksLTTq" />
`;

exports[`test should render the Follow Button 1`] = `
<button
  className="hMPQfj" />
`;
